src/pool.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Execute a bunch of promises in parallel limiting the number of parallel pending promises.
+ *
+ * @param {number} workers Number of parallel workers.
+ * @param {Iterable<() => Promise<T>>} promiseFactories Iterable of promise factories.
+ * @returns {Promise<T[]>}
+ * @template T
+ */
+export function pool(workers, promiseFactories) {
+    if (workers < 1) {
+        throw new Error('Number of workers must be positive');
+    }
+
+    const iterator = indexedIterator(promiseFactories);
+    const promises = [];
+    for (let i = 0; i < workers; i++) {
+        promises.push(worker(iterator));
+    }
+
+    return Promise.all(promises)
+        .then((results) => results
+            .reduce((results, workerResults) => results.concat(workerResults))
```

Comment:
Isn't it equivalent to `results.flat()`?

Reply:
Didn't know about `Array.prototype.flat()`, TIL. Is it widely supported, though?

Otherwise another alternative could be: `[].concat(...results)`.